MSAL/src/public/MSALPublicClientApplication.h

```diff
@@ -34,50 +34,156 @@
 
 #pragma mark -
 #pragma mark acquireToken using Login Hint
 
+
+/*!
+    Acquire a token for a new user using interactive authentication
+ 
+    @param  scopes          Scopes to request from the server, the scopes that come back
+                            can differ from the ones in the original call
+    @param  loginHint       A loginHint (usually an email) to pass to the service at the
+                            beginning of the interactive authentication flow. The user returned
+                            in the completion block is not guaranteed to match the loginHint.
+    @param  completionBlock The completion block that will be called when the authentication
+                            flow completes, or encounters an error.
+ */
 - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
                     loginHint:(NSString *)loginHint
               completionBlock:(MSALCompletionBlock)completionBlock;
 
+/*!
+    Acquire a token for a new user using interactive authentication
+ 
+    @param  scopes          Scopes to request from the server, the scopes that come back
+                            can differ from the ones in the original call
+    @param  loginHint       A loginHint (usually an email) to pass to the service at the
+                            beginning of the interactive authentication flow. The user returned
+                            in the completion block is not guaranteed to match the loginHint.
+    @param  uiBehavior      A specific UI behavior for the interactive authentication flow
+    @param  extraQueryParameters    Key-value pairs to pass to the authentication server during
+                                    the interactive authentication flow.
+    @param  completionBlock The completion block that will be called when the authentication
+                            flow completes, or encounters an error.
+ */
 - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
                     loginHint:(NSString *)loginHint
                    uiBehavior:(MSALUIBehavior)uiBehavior
          extraQueryParameters:(NSDictionary <NSString *, NSString *> *)extraQueryParameters
               completionBlock:(MSALCompletionBlock)completionBlock;
 
+/*!
+    Acquire a token for a new user using interactive authentication
+ 
+    @param  scopes                  Scopes to request from the server, the scopes that come back
+                                    can differ from the ones in the original call
+    @param  additionalScopes        Scopes for the service to ask the user to consent to, will
+                                    not be included in the token returned by the service, but
+                                    if the user consented to those scopes you will be able to
+                                    silently acquire tokens with those scopes later without requiring
+                                    further user interaction
```

Comment:
again, from a newbie perspective, it'd be useful to explain scenarios where you'd want to pass a scope in additionalScopes vs. just passing it in the scopes array.